tools/dependencies.sh

```diff
@@ -7,14 +7,14 @@
 # All our automation scripts read this, so at least we're not duplicating this
 # information across all those.
 
-PG_VERSIONS='12 13 14'
+PG_VERSIONS='12 13 14 15'
 
 CARGO_EDIT=0.11.2
 
 # Keep synchronized with extension/Cargo.toml and `cargo install --version N.N.N cargo-pgx` in Readme.md .
-PGX_VERSION=0.5.4
+PGX_VERSION=0.6.0
 
-RUST_TOOLCHAIN=1.60.0
+RUST_TOOLCHAIN=1.65.0
```

Comment:
Dang, really?  Do they really require 1.65 now? :(

Reply:
Hmm, for some reason they removed rust-version from their Cargo.toml files so it's back to getting confusing and potentially misleading error messages when building with too old a compiler?

Reply:
The `rust-version` in the Cargo.toml files were never accurate anyways. My testing has revealed that it fails on 1.60 and works on 1.65, so I just went the most recent version – I think the actual MSRV might be 1.63 or 1.61 though.

Reply:
pgx 0.6.0 builds fine here on 1.60.0 .

We are starting to fall behind, for sure.  I suggest we update to 1.64.0 in 1.14, not in this release.

Reply:
It [failed in CI on 1.60](https://github.com/timescale/timescaledb-toolkit/actions/runs/3593608416/jobs/6050778282):

```
   Compiling pgx v0.6.0
error[E0658]: use of unstable library feature 'slice_ptr_len'
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pgx-0.6.0/src/slice.rs:28:32
   |
28 |         index.le(&self.pallocd.len()).then(|| self.get_unchecked(index))
   |                                ^^^
   |
   = note: see issue #71146 <https://github.com/rust-lang/rust/issues/71146> for more information
```

and that feature was [partially stabilized in 1.63](https://github.com/rust-lang/rust/pull/94640) – I think 1.63 is the MSRV. I'll do some testing.

Reply:
> The rust-version in the Cargo.toml files were never accurate anyways

They were accurate when they were born:  https://github.com/tcdi/pgx/pull/575

It's a shame to leave dependents to reverse-engineer the requirements by trying various versions.  Maybe we can ask them to bring them back and add verifying them to the release checklist?

Also, I think I tested the wrong thing:  I tested only building cargo-pgx, not pgx.  Whoops!

Updating to 1.64 or 1.63 here is fine by me.

Thanks!

Reply:
I updated the Rust version used to 1.63.